### DISCUSSION TITLE
can i get selected columns value from database of jointjs example

### DISCUSSION THREAD

**Discussion by babyandy0111** [clientIO/joint#2095] [Q&A] [answered]
### Introduction

https://github.com/clientIO/joint/discussions/1682#discussioncomment-2847887

I using database of example from jointjs example and this ref.
I want to save relation between cell of value source and target.  
but i can't get selected value from source and target ... i just saw id and all attributes. 
so what can i do for this requirement? thanks ~

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by jamesgeorgewilliams** [comment, marked as answer]
I am not totally sure what you mean by "selected value".

If you mean the names of the items, you could access them as follows:

```js
paper.on('link:connect', (linkView, _evt, _newEndView, _newEndMagnet, _endType) => {
    const link = linkView.model;
    console.log(link.prop('source')); // {id: '6029c9f7-d3b8-465f-a481-0f3b8b5a60ba', port: 'full_name'}
    console.log(link.prop('target')); // {id: '9cfe0b59-8947-4786-9e91-059e696a7a35', port: 'price'}

    let currentSource;
    let currentTarget;

    linkView.sourceView.model.prop('columns').forEach((column) => {
        if (column.name === link.prop('source').port) {
            currentSource = column;
        }
    });

    linkView.targetView.model.prop('columns').forEach((column) => {
        if (column.name === link.prop('target').port) {
            currentTarget = column;
        }
    });

    console.log(currentSource); // {name: 'full_name', type: 'varchar'}
    console.log(currentTarget); // {name: 'price', type: 'int'}
});
```

**Reply by babyandy0111** [reply]
amazed, I think. I need familiar jointJS more ... 
but anyway ... as your answer this is what I want.
and you solved my problem !
thanks !
